unit-tests/src/test/java/org/eclipse/collections/impl/utility/ListIterateTest.java

```diff
@@ -192,6 +196,30 @@ private void assertCollect(List<Boolean> list)
         Verify.assertListsEqual(newCollection2, FastList.newListWith("true", "false", "null"));
     }
 
+    /**
+     * @since 9.1.
+     */
+    @Test
+    public void collectWithIndex()
+    {
+        MutableList<Boolean> list = Lists.fixedSize.of(true, false, null);
```

Comment:
Can you please add 1 more test case which uses ArrayList<>()?

Reply:
Yes, I will add a test.